### ISSUE TITLE
Tracking issue for deprecating serde_macros

### ISSUE THREAD

**Issue by dtolnay** [serde-rs/serde#545]
Serde_derive has many advantages over serde_macros and a short list of disadvantages which are within sight of being eliminated.

Here are the things that need to happen before serde_derive can replace serde_macros:

- [x] Documentation https://github.com/serde-rs/serde-rs.github.io/issues/21
- [x] Macros 1.1 bug https://github.com/rust-lang/rust/issues/36211 (we have a workaround)
- [x] Compilation time https://github.com/serde-rs/serde/issues/540
- [x] Dependency nightmare (depending on how we resolve compile time) https://github.com/serde-rs/serde/issues/543

Nice-to-haves but not blockers:

- [x] Error spans are misleading https://github.com/rust-lang/rust/issues/36218

**Comment by SShrike** [comment]
>Serde_derive has many advantages over serde_derive

Shouldn't that be "Serde_derive has many advantages over serde_macros" or something?

**Comment by dtolnay** [comment]
Fixed, thanks.

**Comment by dtolnay** [comment]
I released [0.8.10](https://github.com/serde-rs/serde/releases/tag/0.8.10) announcing the deprecation.

**Comment by dtolnay** [comment]
I made a [users.rust-lang.org post](https://users.rust-lang.org/t/serde-transitioning-to-macros-1-1/7437?u=dtolnay) as well.
